pkg/main.go

```diff
@@ -103,6 +107,14 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "HypershiftDeployment")
 		os.Exit(1)
 	}
+
+	if err = (&autoimport.Reconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
```

Comment:
@ianzhang366 I guess this will put the ManagedCluster CRD requirement on using HypershiftDeployment (similar to the ManifestWork.

Reply:
yup... this would make the `hypershift-deployment` closer couple with ACM

Reply:
yeah, but I think it is okay. isn't it?

Reply:
if we only will use this operator inside acm, then yes. if we want other user to use the hypershift-deployment as a beginner tool for exploring the hypershift, then it's not good... since we bring extra concept for a beginner.

Reply:
OK, So I think I can add another PR to make this `auto-import` controller a feature gate, and keep it enabled as default. If users don't want this feature, then they can disable it. @jnpacker @ianzhang366 WDYT?

Reply:
Or keep it disabled by default, and we just enable it in MCE.

Reply:
I lost track about why don't we put import logic into our import-controller?

Reply:
Import-controller is not responsible for creating the managed cluster, it only handles the logic after it is created. The managed cluster is created by UI, clusterclaim-controller, or users. 
And if we put this logic in import-controller, import controller also needs to watch the new CR, HypershiftDeployment.

cc @qiujian16

Reply:
I'm in favor of putting this to import-controller for now, when we have the UI for hypershift, then we can drop the logic of creating the managedCluster CR from the import-controller and use UI for creating the managedCluster CR.

in this way, the implementation algin with the hive, I mean, we don't have our import logic in hive, instead the import-controller watches for hive's resource and acts upon the hive resources.

what do you say?

Reply:
I have a different opinion on putting this in import-controller right now. If it already has the deps on manifestwork, I think it is ok to have deps on managedcluster. you can have a flag like --auto-import in the controller to enable this controller. We need to think about it more carefully before putting this logic into import controller, since import controller is a pretty critical component. We have a claim-controller that wire provisioner API to managedcluster. This controller might do a similar thing as claim-controller.

Reply:
Long term, we might consider putting this into 3 pieces. 

multiple provisioner controller --> controller that wires provisioner to managedcluster -> registration

today the `wire` controller actually relates to multiple controllers including claim controller, import controller etc. We might think about having a unified one to do that.

Reply:
Yeah, agree with qiujian, For hive, the cluster claim case, there is [clusterclaim-controller](https://github.com/stolostron/clusterclaims-controller/blob/main/controllers/clusterclaims/clusterclaims-controller.go) to watch clusterclaim and create managedcluster.
And think about modularity, put it here, if we want to disabled all hypershift related stuff in MCE, we just need to delete the hypershift-deployment-controller and hypershift-addon, put it in import-controller, we also need to figure out a way to disable the logic in import controller. 

So I prefer to put it here, and then use the feature gate control, disabled by default.